docs/components/templating-and-jsx.md

```diff
@@ -248,6 +248,48 @@ ensure that each child has a key which does not change and which is unique
 among all its siblings.
 :::
 
+### Automatic Key Insertion
+
+During compilation Stencil will automatically add key attributes to any JSX
+nodes in your component's render method which are not nested within curly
+braces. This allows Stencil’s runtime to accurately reconcile children when
+their order changes or when a child is conditionally rendered.
+
+For instance, consider a render method looking something like this:
+
+```tsx
+  render() {
+    return (
+      <div>
+        {this.disabled && (
+          <div />
+        )}
```

Comment:
I think we could strengthen this section a little bit by making it clearer that it's in curly braces whereas the "slot-wrapper" is not.

I think we could accomplish that by:
1. Adding whitespace (newlines) after the opening `{` and before the closing `}`
2. Adding a text node and/or id to the div that helps draw attention to it being in curlies:

```suggestion
        {
          this.disabled && <div id="in-curly-braces">Won't receive automatic key insertion</div>
        }
```

☝️ That might not be the _best_ in terms of presentation (particularly may be too wide for the viewport), but is the general idea. Thoughts?

Reply:
mmm I think I'll go for something like

```tsx
  { this.disabled && <div id="no-key">no key!</div> }
```

just to make sure it's narrow enough